src/lib/y2partitioner/actions/controllers/partition_table.rb

```diff
@@ -0,0 +1,108 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2partitioner/device_graphs"
+require "y2partitioner/ui_state"
+
+module Y2Partitioner
+  module Actions
+    module Controllers
+      # This class stores information about a future partition table so that
+      # information can be shared across the different dialogs of the
+      # process. It also takes care of updating the devicegraph when needed.
+      class PartitionTable
+        include Yast::I18n
+        include Yast::Logger
+
+        # Partition table type to be used. If not set, the default type for
+        # this type of disk is used: one of Gpt, Msdos, Dasd.
+        #
+        # @return [Y2Storage::PartitionTables::Type]
+        attr_accessor :type
+
+        # New partition table created by the controller.
+        #
+        # @return [Y2Storage::PartitionTables::Any]
```

Comment:
Well, not really; ::Base is the one thing that is guaranteed never to be returned here since that's only an abstract base class. It is in fact always a subclass of ::Base.